client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -581,6 +588,22 @@ export class CheckoutThankYou extends Component<
 				pageContent = <DomainOnlyThankYou purchases={ purchases } receiptId={ receiptId } />;
 			} else if ( purchases.length === 1 && isPlan( purchases[ 0 ] ) ) {
 				pageContent = <PlanOnlyThankYou primaryPurchase={ purchases[ 0 ] } />;
+			} else if ( wasTitanEmailOnlyProduct ) {
+				pageContent = (
+					<TitanSetUpThankYou
+						domainName={ purchases[ 0 ].meta }
+						emailAddress={ email }
+						isDomainOnlySite={ this.props.domainOnlySiteFlow }
+					/>
+				);
+			} else if ( isTitanWithoutMailboxes( selectedFeature ) && domainPurchase ) {
+				// To reproduce it: https://github.com/Automattic/wp-calypso/pull/87098#titan-wo-mailboxes
```

Comment:
```suggestion
				// Users may purchase Titan subscription without specifying the mailbox name using
				// the onboard with email flow (https://wordpress.com/start/onboarding-with-email/mailbox-domain)
```

I think it's clearer to describe the scenario rather than pointing to the PR.